src/app/Caches/Create/CreateCacheWizard.tsx

```diff
@@ -0,0 +1,212 @@
+
+  // const optionsStep = {
+  //   name: showCreateStep
+  //     ? `${createStepRadio} Options`
+  //     : `${updateStepRadio} Options`,
+  //   steps: [
+  //     {
+  //       name: 'Substep 1',
+  //       component: 'Substep 1',
+  //     },
+  //     {
+  //       name: 'Substep 2',
+  //       component: 'Substep 2',
+  //     },
+  //     {
+  //       name: 'Substep 3',
+  //       component: 'Substep 3',
+  //     },
+  //   ],
+  // };
+
+  const steps = [
+    getStartedStep,
+    ...(showWithoutEditorStep ? [configureWithoutEditorStep] : []),
+    ...(showWithEditorStep ? [configWithEditorStep] : []),
+    // ...(showOptionsStep ? [optionsStep] : []),
+  ];
+
+  const title = 'Progressive wizard';
+  const closeWizard = () => {
+    console.log('close wizard');
+  };
+
+  const onGoToStep = (name) => {
+    console.log('TEST', name);
+
+    // Remove steps after the currently clicked step
+    if (name === 'Get started') {
+      setShowOptionsStep(false);
+    } else if (name === 'Create options' || name === 'Update options') {
+      setShowWithEditorStep(false);
+      setShowWithoutEditorStep(false);
+    }
+  };
+
+  const getNextStep = (activeStep) => {
+    if (activeStep.name === 'Get started') {
+      if (getStartedStepRadio === 'radio1') {
+        setShowWithEditorStep(false);
+        setShowWithoutEditorStep(true);
+      } else {
+        setShowWithEditorStep(true);
+        setShowWithoutEditorStep(false);
+      }
+    }
+  };
+
+  //   const getPreviousStep = (activeStep, callback) => {
+  //     if (activeStep.name === 'Review') {
+  //       this.setState(
+  //         {
+  //           showReviewStep: false,
+  //         },
+  //         () => {
+  //           callback();
+  //         }
+  //       );
+  //     } else if (activeStep.name === 'Substep 1') {
+  //       this.setState(
+  //         {
+  //           showOptionsStep: false,
+  //         },
+  //         () => {
+  //           callback();
+  //         }
+  //       );
+  //     } else if (activeStep.name === 'Create options') {
+  //       this.setState(
+  //         {
+  //           showCreateStep: false,
+  //         },
+  //         () => {
+  //           callback();
+  //         }
+  //       );
+  //     } else if (activeStep.name === 'Update options') {
+  //       this.setState(
+  //         {
+  //           showUpdateStep: false,
+  //         },
+  //         () => {
+  //           callback();
+  //         }
+  //       );
+  //     } else {
+  //       callback();
+  //     }
+  //   };
+  // };
+
+  const CustomFooter = (
+    <WizardFooter>
+      <WizardContextConsumer>
+        {({
+          activeStep,
+          goToStepByName,
+          goToStepById,
+          onNext,
+          onBack,
+          onClose,
+        }) => {
+          return (
+            <>
+              <Button
+                variant="primary"
+                type="submit"
+                onClick={() => {
+                  // onNext();
```

Comment:
@karesti How can we use onNext() function inside useEffect()?

Reply:
you can directly call the method from useEffect. what are you trying to do?

Reply:
I am not sure how to pass that function in useEffect